code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,413 @@
+	var/allow_items = TRUE
+
+/obj/machinery/computer/cryopod/Initialize()
+	. = ..()
+	GLOB.cryopod_computers += src
+
+/obj/machinery/computer/cryopod/Destroy()
+	GLOB.cryopod_computers -= src
+	..()
+
+/obj/machinery/computer/cryopod/update_icon_state()
+	if(machine_stat & (NOPOWER|BROKEN))
+		icon_state = "cellconsole"
+		return ..()
+	icon_state = "cellconsole_1"
+	return ..()
+
+/obj/machinery/computer/cryopod/ui_interact(mob/user, datum/tgui/ui)
+	if(machine_stat & (NOPOWER|BROKEN))
+		return
+
+	add_fingerprint(user)
+
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "CryopodConsole", name)
+		ui.open()
+
+/obj/machinery/computer/cryopod/ui_data(mob/user)
+	var/list/data = list()
+	data["allow_items"] = allow_items
+	data["frozen_crew"] = frozen_crew
+	data["frozen_items"] = list()
+
+	if(allow_items)
+		data["frozen_items"] = frozen_items
+
+	var/obj/item/card/id/id_card
+	var/datum/bank_account/current_user
+	if(isliving(user))
+		var/mob/living/person = user
+		id_card = person.get_idcard()
+	if(id_card?.registered_account)
+		current_user = id_card.registered_account
+	if(current_user)
+		data["account_name"] = current_user.account_holder
+
+	return data
+
+/obj/machinery/computer/cryopod/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/user = usr
+
+	add_fingerprint(user)
+
+	switch(action)
+		if("one_item")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
+			if(!allow_items) return
+
+			if(!params["item"])
+				return
+
+			var/obj/item/item = frozen_items[text2num(params["item"])]
+			if(!item)
+				to_chat(user, "<span class='notice'>\The [item] is no longer in storage.</span>")
+				return
+
+			visible_message("<span class='notice'>The console beeps happily as it disgorges \the [item].</span>")
+			item.forceMove(get_turf(src))
+			frozen_items -= item
+
+		if("all_items")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
+			if(!allow_items) return
+
+			visible_message("<span class='notice'>The console beeps happily as it disgorges the desired objects.</span>")
+
+			for(var/obj/item/item in frozen_items)
+				item.forceMove(get_turf(src))
+				frozen_items -= item
+
+	return TRUE
+
+// Cryopods themselves.
+/obj/machinery/cryopod
+	name = "cryogenic freezer"
+	desc = "Suited for Cyborgs and Humanoids, the pod is a safe place for personnel affected by the Space Sleep Disorder to get some rest."
+	icon = 'icons/obj/machines/cryopod.dmi'
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 3 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 3 MINUTES // This is reduced to 30 seconds if a player manually enters cryo
+	var/despawn_world_time = null          // Used to keep track of the safe period.
```

Comment:
```suggestion
	var/despawn_world_time = null // Used to keep track of the safe period.
```

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#tabs-not-spaces